tags/TagMapContext.md

```diff
@@ -0,0 +1,119 @@
+# Summary
+A `Tag` is a key-value pair that can be used to label anything that is associated
+with a specific operation, such as an HTTP request. These `Tag`s are used to
+aggregate measurements in a [`View`](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/DataAggregation.md#view) 
+according to unique value of the `Tag`s. The `Tag`s can also be used to filter (include/exclude)
+measurements in a `View`. `Tag`s can further be used for logging and tracing.
+
+# TagMap 
+`TagMap` is an abstract data type that represents collection of tags. 
+i.e., each key is associated with exactly one value, but multiple keys can be
+associated with the same value.  `TagMap` is serializable, and it represents
+all of the information that could be propagated inside the process and across process boundaries.  
+`TagMap` is a recommended name but languages can have more language specific name.
+
+## TagKey
+
+`TagKey` consists of source, metadata and name.
+
+### Source
+
+Source of a tag is either remote or local.
+ 
+**Local Source**
```

Comment:
who decides whether local tags should be propagated? Should there be "local-only" tags?